stackslib/src/net/download/nakamoto.rs

```diff
@@ -0,0 +1,4283 @@
+                .into_iter()
+                .chain(prev_wts.into_iter())
+                .chain(cur_wts.into_iter())
+            {
+                test_debug!("Consider wanted tenure: {:?}", &wt);
+                let wt_rc = sortdb
+                    .pox_constants
+                    .block_height_to_reward_cycle(sortdb.first_block_height, wt.burn_height)
+                    .expect("FATAL: height before system start");
+                if wt_rc + 1 == sort_rc {
+                    prev_wanted_tenures.push(wt);
+                } else if wt_rc == sort_rc {
+                    cur_wanted_tenures.push(wt);
+                } else {
+                    test_debug!("Drop wanted tenure: {:?}", &wt);
+                }
+            }
+
+            prev_wanted_tenures.sort_by(|wt1, wt2| wt1.burn_height.cmp(&wt2.burn_height));
+            cur_wanted_tenures.sort_by(|wt1, wt2| wt1.burn_height.cmp(&wt2.burn_height));
+
+            test_debug!("prev_wanted_tenures is now {:?}", &prev_wanted_tenures);
+            test_debug!("wanted_tenures is now {:?}", &cur_wanted_tenures);
+
+            self.prev_wanted_tenures = if prev_wanted_tenures.len() > 0 {
+                Some(prev_wanted_tenures)
+            } else {
+                None
+            };
+            self.wanted_tenures = cur_wanted_tenures;
+            self.reward_cycle = sort_rc;
+        } else {
+            test_debug!(
+                "Append {} wanted tenures: {:?}",
+                new_wanted_tenures.len(),
+                &new_wanted_tenures
+            );
+            self.wanted_tenures.append(&mut new_wanted_tenures);
+            test_debug!("wanted_tenures is now {:?}", &self.wanted_tenures);
+        }
+
+        Ok(())
+    }
+
+    /// Initialize `self.wanted_tenures` and `self.prev_wanted_tenures` for the first time, if they
+    /// are not set up yet.  At all times, `self.prev_wanted_tenures` ought to be initialized to the last
+    /// full reward cycle's tenures, and `self.wanted_tenures` ought to be initialized to the
+    /// ongoing reward cycle's tenures.
+    pub(crate) fn initialize_wanted_tenures(
+        &mut self,
+        sort_tip: &BlockSnapshot,
+        sortdb: &SortitionDB,
+    ) -> Result<(), NetError> {
+        // check for reorgs
+        let reorg = PeerNetwork::is_reorg(self.last_sort_tip.as_ref(), sort_tip, sortdb);
+        if reorg {
+            // force a reload
+            test_debug!("Detected reorg! Refreshing wanted tenures");
+            self.prev_wanted_tenures = None;
+            self.wanted_tenures.clear();
+        }
+
+        if self
+            .prev_wanted_tenures
+            .as_ref()
+            .map(|pwts| pwts.len())
+            .unwrap_or(0)
+            < usize::try_from(sortdb.pox_constants.reward_cycle_length)
+                .expect("FATAL: usize cannot support reward cycle length")
+        {
+            // this is the first-ever pass, so load up the last full reward cycle
```

Comment:
Does this need to check if this is the first reward cycle in nakamoto, so the prev_wanted_tenures should be `None`?

Reply:
`prev_wanted_tenures` should never be `None`, even if it represents only epoch 2.x sortitions.